examples/handle-head-request/src/main.rs

```diff
@@ -0,0 +1,84 @@
+//! Run with
+//!
+//! ```not_rust
+//! cd examples && cargo run -p example-handle-head-reqeust
+//! ```
+
+use axum::response::{IntoResponse, Response};
+use axum::{http, routing::get, Router};
+use std::net::SocketAddr;
+
+fn app() -> Router {
+    Router::new().route("/get-head", get(get_head_handler))
+}
+
+#[tokio::main]
+async fn main() {
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+
+    axum::Server::bind(&addr)
+        .serve(app().into_make_service())
+        .await
+        .unwrap();
+}
+
+// get routes will also be called for HEAD requests but will have the response body removed
+// can handle implicit head method request by extract `http::Method` from request
+async fn get_head_handler(method: http::Method) -> Response {
+    if method == http::Method::HEAD {
+        return ([("x-some-header", "header from HEAD")]).into_response();
+    }
+
+    ([("x-some-header", "header from GET")], "body from GET").into_response()
```

Comment:
Can we have something a little more elaborate here? Or at least a comment above this line that you would usually only special-case `HEAD` if computing the body has some relevant compute cost?